org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/XtextBuilder.java

```diff
@@ -261,7 +267,22 @@ public boolean isCanceled() {
 			task.stop();
 			Job.getJobManager().removeJobChangeListener(MAKE_EGIT_JOB_SYSTEM);
 		}
-		return getProject().getReferencedProjects();
+		return getReferencedProjects();
+	}
+
+	/**
+	 * @since 2.19
+	 */
+	protected IProject[] getReferencedProjects() throws CoreException {
+		interestingProjects = toBeBuiltComputer.getInterestingProjects(getProject());
+		return interestingProjects.toArray(new IProject[0]); 
```

Comment:
unusal pattern. know that better als list.toArray(new Type[list.size])

Reply:
https://shipilev.net/blog/2016/arrays-wisdom-ancients/